types/validator/validator.go

```diff
@@ -131,8 +131,8 @@ func (val *Validator) IncSequence() {
 }
 
 // UpdateLastJoinedHeight updates the last height at which the validator joined the committee.
-func (val *Validator) UpdateLastJoinedHeight(height uint32) {
-	val.data.LastJoinedHeight = height
+func (val *Validator) UpdateLastSortitionHeight(height uint32) {
```

Comment:
what is your opinion about below idea?
should we use mutex for `UpdateLastSortitionHeight` method? because during our update maybe `LastSortitionHeight` method call so the result of `LastSortitionHeight` is not correct.

Reply:
@b00f 
@themantre 

what is your opinion?

Reply:
Adding a locking mechanism such as a mutex to low-level types like `account` or `validator` can have some side effects:

-  It could slow down the system, as these types are widely used in other packages.
- It can result in many locks, making the system difficult to debug.

A better approach is to ensure that the main packages are thread-safe and that resources are not shared between them. 
When sharing is unavoidable, we pass a copy of the underlying object to prevent shared resource usage.
As an example look the [validator](https://github.com/pactus-project/pactus/blob/6c36120f5f1958f58412c3ccb63ba0b1671bbcf4/sandbox/sandbox.go#L149) function in sandbox and the way we clone the validator objects there.